fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/temp/properties/CreatesDuplicatesProperty.java

```diff
@@ -104,4 +113,12 @@ public static boolean evaluate(@Nonnull RelationalExpression expression, @Nonnul
         // Won't actually be null for relational planner expressions.
         return expression.acceptPropertyVisitor(new CreatesDuplicatesProperty(context));
     }
+
+    @Nonnull
+    @Override
+    @SpotBugsSuppressWarnings("NP_PARAMETER_MUST_BE_NONNULL_BUT_MARKED_AS_NULLABLE")
+    public Boolean evaluateAtQuantifier(@Nonnull final Quantifier quantifier, @Nullable final Boolean rangesOverResult) {
+        // since we do visit expression references in this property we can insist on rangesOverResult not being null
```

Comment:
```suggestion
        // since we visit the expression reference under the quantifier, and don't return null ourselves, we can
        // insist that rangesOverResult is never null
```